js/blog-functions.js

```diff
@@ -0,0 +1,123 @@
+var blog = {};
+
+blog.authorFromDB = function() {
+  $('.articleAuthor').children('option:nth-child(n+2)').remove();
+  webDB.execute(
+    'SELECT DISTINCT author FROM articles ORDER BY author'
+    , function(resultArray) {
+      blog.uniqueAuthors = resultArray;
+      resultArray.forEach(function(ele) {
+        $('.articleAuthor').append('<option>' + ele.author + '</option>');
+      });
+    }
+  );
+};
+
+blog.categoryFromDB = function() {
+  $('.articleCategory').children('option:nth-child(n+2)').remove();
+  webDB.execute(
+    'SELECT DISTINCT category FROM articles ORDER BY category'
+    , function(resultArray) {
+      blog.uniqueCategories = resultArray;
+      resultArray.forEach(function(ele) {
+        $('.articleCategory').append('<option>' + ele.category + '</option>');
+      });
+    }
+  );
+};
+
+blog.titleFromDB = function() {
+  $('.articleTitle').children('option:nth-child(n+2)').remove();
+  webDB.execute(
+    'SELECT DISTINCT title FROM articles ORDER BY title'
+    , function(resultArray) {
+      blog.uniqueTitles = resultArray;
+      resultArray.forEach(function(ele) {
+        $('.articleTitle').append('<option>' + ele.title + '</option>');
+      });
+    }
+  );
+};
+
```

Comment:
You probably don't want to be re-executing SQL queries every time you use a selector. At a minimum, this is causing your filters to re-append selections every time a user makes a selection, causing repetition. 

![screen shot 2015-12-28 at 10 57 39 am](https://cloud.githubusercontent.com/assets/3937302/12023779/df63a518-ad51-11e5-9f0d-807bc3d65f28.png)